src/reconstruction/comparison/modelBorgifier/src/compareAllMets.m

```diff
@@ -0,0 +1,194 @@
+% This file is published under Creative Commons BY-NC-SA.
+%
+% Please cite:
+% Sauls, J. T., & Buescher, J. M. (2014). Assimilating genome-scale 
+% metabolic reconstructions with modelBorgifier. Bioinformatics 
+% (Oxford, England), 30(7), 1036?8. http://doi.org/10.1093/bioinformatics/btt747
+%
+% Correspondance:
+% johntsauls@gmail.com
+%
+% Developed at:
+% BRAIN Aktiengesellschaft
+% Microbial Production Technologies Unit
+% Quantitative Biology and Sequencing Platform
+% Darmstaeter Str. 34-36
+% 64673 Zwingenberg, Germany
+% www.brain-biotech.de
+%
+function metScores = compareAllMets(Cmodel, Tmodel)
+% compareAllMets compares and scores similarity between all pairwise
+% combinations of metabolites between two models. It is not however used
+% by the main workflow. 
+%
+% USAGE:
+%    metScores = compareAllMets(Cmodel,Tmodel
+%
+% INPUTS:
+%    Cmodel:    Comparison model. 
+%    Tmodel:    Template model. 
+%
+% OUTPUTS:
+%    metScore:  Matrix of scores of all pairwise comparisons. 
+%
+% CALLS:
+%    None
+%
+% CALLED BY:
+%    None
+
+
+%% Declare variables and scoring structure.
+nCmets = length(Cmodel.mets) ;
+nTmets = length(Tmodel.mets) ;
+
+% Score values.
+ScoreVal = struct('ID', [10,-1], ...
+                  'compartment', [0,-100], ...
+                  'name', [10,-1], ...
+                  'formula', [10,-5], ...
+                  'charge', [1,-1], ...
+                  'KEGGID', [10,-5], ...
+                  'SEEDID', [10,-5], ...
+                  'ChEBIID', [10,-5], ...
+                  'PubChemID', [10,-5], ...
+                  'InChIString', [10,-5]) ;
+
+% Score matrix.
+metScores = zeros(nCmets, nTmets) ;
+
+% Compartments of every met in T, used later for comparison.
+tComp = cell(nTmets, 1) ;
+for iMet = 1:nTmets
+    tComp{iMet} = Tmodel.mets{iMet}(end - 2:end) ;
+end
+
+
+%% Compare cMet to tMets
+h = waitbar(0, 'Comparing metabolites') ;
+for iMet = 1:nCmets
+    % metsID Match. There is always one ID.
+    % Match name, removing '[.]' and add regular expression info. 
+    if ~isempty(regexp(Cmodel.mets{iMet}, '\[.\]', 'once')) 
```

Comment:
this regexp is questionable, as metabolite compartments are not necessarily restricted to one symbol . 
I would suggest using:
Cmodel.mets{iMet} = strcat('(\||\<)', regexprep(Cmodel.mets{iMet} ,'[.*?\]$',''), '(\>|\[|_|\|)');

Since we assume that the metabolite id is followed by the location.